tests/platform_tests/api/test_thermal.py

```diff
@@ -28,7 +28,15 @@
     pytest.mark.topology('any')
 ]
 
+@pytest.fixture(scope="class")
+def gather_facts(request, duthost):
+    # Get platform facts from platform.json file
+    request.cls.chassis_facts = duthost.facts.get("chassis")
+    if not request.cls.chassis_facts:
+        logger.warning("Unable to get chassis_facts from platform.json, test results will not be comprehensive")
 
+
+@pytest.mark.usefixtures("gather_facts")
 class TestThermalApi(PlatformApiTestBase):
 
     num_thermals = None
```

Comment:
Suggest declaring `chassis_facts = None` here for completeness.

Reply:
fixed